server/src/Commands.def

```diff
@@ -12,7 +12,9 @@
 #error Please define the 'X(id, command, description)' macro before inclusion!
 #endif
 
-X(Candidates, "candidates", "print completion candidates (require previous complete)")
+X(Candidates, "candidates",
+  "PREFIX STYLE - print completion candidates (require previous complete). "
```

Comment:
Sounds reasonable. 

> in Elisp: 'exact, 'case-insensitive and 'smart-case

So need we create a new variable in elisp or just reuse company-irony-ignore-case?

> in C++ use an enum class for std::string caseStyle, so you benefit more from the type system, only when read/validate the command you deal with a string

So we need create a new enum like 'enum Style' (or other name if you have better option :)) and a new StyleConverter to validate it. Both irony.cpp and command.cpp need access this enum. Is it ok to create a new header file for this enum?